test/metric_value_benchmark/eks_daemonset_test.go

```diff
@@ -66,19 +76,80 @@ func (e *EKSDaemonTestRunner) validateInstanceMetrics(name string) status.TestRe
 	return testResult
 }
 
-func (e *EKSDaemonTestRunner) GetTestName() string {
+func (e *EKSDaemonTestRunner) validateLogs(eks *environment.MetaData) status.TestResult {
+	testResult := status.TestResult{
+		Name:   "emf-logs",
+		Status: status.FAILED,
+	}
+
+	now := time.Now()
+	group := fmt.Sprintf("/aws/containerinsights/%s/performance", eks.EKSClusterName)
+
+	// need to get the instances used for the EKS cluster
+	eKSInstances, err := awsservice.GetEKSInstances(eks.EKSClusterName)
+	if err != nil {
+		log.Println("failed to get EKS instances", err)
+		return testResult
+	}
+
+	for _, instance := range eKSInstances {
+		validateLogContents := func(s string) bool {
+			return strings.Contains(s, fmt.Sprintf("\"ClusterName\":\"%s\"", eks.EKSClusterName))
+		}
+
+		var ok bool
+		stream := *instance.InstanceName
+		ok, err = awsservice.ValidateLogs(group, stream, nil, &now, func(logs []string) bool {
+			if len(logs) < 1 {
+				log.Println(fmt.Sprintf("failed to get logs for instance: %s", stream))
+				return false
+			}
+
+			for _, l := range logs {
+				var eksClusterType awsservice.EKSClusterType
+				err := json.Unmarshal([]byte(l), &eksClusterType)
+				if err != nil {
+					log.Println("failed to unmarshal log file")
+				}
+
+				log.Println(fmt.Sprintf("eksClusterType is: %s", eksClusterType.Type))
```

Comment:
These arent really ClusterTypes... they are just LogEvent types

Reply:
I can clean this up at some point. Will add it to my list of things